### DISCUSSION TITLE
HyperHDR won´t connect to HyperSerial via USB/adalight

### DISCUSSION THREAD

**Discussion by Callahan78** [awawa-dev/HyperHDR#97] [Q&A] [answered]
Hi,
I tried since a few days but I can´t get it working . . .
Have a Raspberry Pi 4b with HyperHDR which is connected to a ESP32 with HyperSerialWLED. Via Wifi everything is running fine (SK6812).
But when I connect via USB there no connection will be established.
I made all the recommended settings in HyperHDR (Baudrate 2000000, AWA protocol enabled ...).
No errors in log.
Any ideas what I could try?
Best regards

**Comment by awawa-dev** [comment, marked as answer]
Hi
In 99% of cases it turns out that ESP doesn't have 2Mb capable chip onboard (for example cp2104, CH340G ...it's written in the requirements). So first thing you should do is to verify it on your board. One chip should have similar marking. If it's popular cp2102 it won't work as it's only 1Mb capable.
Regards
Awawa

**Comment by Callahan78** [comment]
OK, I wish I had ask some days earlier . . . it´s a CP2102 Chip. So I will order another.
Will "D1 Mini NodeMcu mit ESP8266-12F WLAN Module CH340G" do the job?

**Comment by awawa-dev** [comment]
ESP8266 wemos D1 mini CH340G works fine. I use it with my setup. It's a bit a pain to provide a system driver for Windows and older macOS but under Linux it works out of box.

**Reply by Callahan78** [reply]
Ok, I´ll try  . . . . :-)
Thank you very much for the quick answers!

**Comment by Callahan78** [comment]
Back from holidays and in the meantime my "ESP8266-12F WLAN Module CH340G" has arrived. I flashed it with "WLED_0.13.0-b2_ESP8266.bin".
But same here: works fine via Wifi (type "wled") but not with USB (type "adalight").
Any ideas?

By the way: in Wifi-Mode the ESP´s web interface is not reachable while HyperHDR is accessing it ("Connection to light failed!"). Is that correct? With my former ESP (CP2102 Chip) the web interface stays accessible.

**Comment by awawa-dev** [comment]
If you flashed it with the official WLED release then you can't use 2MB awa protocol. You can use only 115200 adalight with disabled awa protocol. Make sure that it works first with such settings.

**Reply by Callahan78** [reply]
Damn . . . I screwed up the links . . . you´re right. Now I flashed with "HyperSerialEsp8266.ino.d1_mini_RGBW_FIRSTLED_ENABLED.bin".
But now there is no WLED-AP to configure the device. I reflashed sometimes without any error, but no WLED-AP after flashing.
Sorry for my stupidness . . . .

**Comment by awawa-dev** [comment]
You should stay with the official WLED as you've already flashed it first and as I suggested. It's not a normal behavior for ESP device, maybe something is wrong with the flash procedure or in the flash memory itself..sometimes WLED-AP doesn't not appear after an upgrade because it remembers some things and the device www panel IP/address name depends then on the router assignment.

**Comment by awawa-dev** [comment]
````HyperSerialEsp8266.ino.d1_mini_RGBW_FIRSTLED_ENABLED.bin```` is not a HyperSerialWLED...it doesnt have a WiFi interface anyway ;)

**Comment by awawa-dev** [comment]
You've probably searched for that: https://github.com/awawa-dev/HyperSerialWLED/releases/tag/v3.0.0.0

**Reply by Callahan78** [reply]
Ok, now I am completely confused . . . . :-p
Which image should I flash for my purpose (SK6812RGBW > ESP8266-12F >USB> HyperHDR)?

**Reply by Callahan78** [reply]
Just tested with "nodemcuv2_ESP8266.bin", WLED-AP available, Wifi connect works, USB connect doesn´t

**Comment by awawa-dev** [comment]
OK, there are two different solution to have 2Mb transmission if you have capable ESP with CH340G:
- HyperSerialEsp8266 with no WiFi interface, fast, pure bridge for awa protocol for ws281x and sk6812 (with a calibration for cold or neutral). It has nothing common with a WLED project.
- HyperSerialWLED, fork of WLED with 2Mb awa protocol implemented.

Just to avoid the confusion because I don't know what are your settings now ... 

Please first test your setup with the original WLED 0.13 to verify that everything is working OK.
Set speed to **115200** and **disable awa protocol**.
If that is working then proceed with the next step.

Flash nodemcuv2_ESP8266.bin from https://github.com/awawa-dev/HyperSerialWLED/releases/tag/v3.0.0.0 
Set speed to **2000000** and **enable awa protocol**.
If it doesn't work please provide full logs from that configuration (upload them on pastebin service).

**Reply by awawa-dev** [reply]
And did you verify your new board if there is a CH340G chip?

**Reply by Callahan78** [reply]
Step 1 worked partially . . . for the 1st time ever I had a USB connection ("WLED is receiving live USB Adalight/TPM2 data"), but a lot of the LEDs where flickering wild.
Nevertheless I continued with step 2, but then again: works in WLED/Wifi-Mode and no connect via USB . . .
Shall I try a new ESP maybe?

Is there a way to proof if it is a CH340G chip? There is no hint on the ESP itself, I just trusted the spec sheet.

**Reply by awawa-dev** [reply]
You must first make sure that the default Adalight mode works with WLED 0.13 at speed of 115200.
With such slow speed it doesn't matter if it's CH340G or not: every serial chip should work with that speed.

If your LED strip is working OK while running with HyperHDR over Wifi then we can exclude problems with a connection ESP to the LED strip (or interference). So maybe the problem is the USB cable?

The only way to tell if it's 340G or not is to verify it with own eyes.
It looks like large black rectangle with 340g marking:
![obraz](https://user-images.githubusercontent.com/69086569/130134372-70d83045-14ec-4a63-a656-0f0c2cc3ffdf.png)
The other side wemos d1:
![obraz](https://user-images.githubusercontent.com/69086569/130134492-a19a8c20-a3b8-4757-b2d5-c763f5e94f69.png)

**Reply by awawa-dev** [reply]
Don't remember if you have a common ground between LED strip and ESP/Raspberry pi?

**Reply by Callahan78** [reply]
Ok, I will reflash with WLED 0.13 and try with another USB cable. Are there some special restrictions/limitations regarding USB cable length?

There is nothing in the chip
![image](https://user-images.githubusercontent.com/88541621/130137601-ee3f35aa-fc1a-4bb1-9864-073b41ad4253.png)
![image](https://user-images.githubusercontent.com/88541621/130137721-73636176-d09a-40a1-b09a-edef794b828b.png)

**Reply by Callahan78** [reply]
> Don't remember if you have a common ground between LED strip and ESP/Raspberry pi?

LED strip and ESP have a common ground. Raspberry Pi is powered by its own power supply.

**Reply by awawa-dev** [reply]
Looks like 340G, view at him at different angles...at some point the markings should be barely visible.
USB2.0 works fine at quite long distance. Never had a problem even over 2M but it depends on the cable quality...or if it's broken or not. Or if your Rpi has enough power to provide a current for Esp...

EDIT

**Reply by Callahan78** [reply]
I looked at it really closely, but I couldn't find anything . . . the chip is completely blank/black.

I just flashed WLED and tried different USB cables. Result is that USB connect could be established, but LEDs are flickering.

It´s during a freezed frame:
https://user-images.githubusercontent.com/88541621/130150208-57cf22cf-8b0b-40fd-9a7a-9a280cd46a07.mp4

Via WLED/Wifi there is no flickering.

![image](https://user-images.githubusercontent.com/88541621/130150387-8d1ec48a-7ba1-4962-ab37-a8e13698411f.png)

**Reply by Callahan78** [reply]
... and sometimes it falls back to the pattern/color which is set in WLED, so it seemes that the connection is lost. After 1-3 seconds it "reconnects" do HyperHDR.

**Reply by awawa-dev** [reply]
Don't know if something from WLED is interfering, in theory it should not. Just disable 'power limit' if you have a proper power supply.

Please provide logs from HyperHDR when that happens. And verify your ````dmesg```` command because maybe there are more serious issues. HyperHDR doesn't have an ability to reconnect serial port connection. 

And don't test with LED strip on the roll please ;) It's a very easy way to make a short circuit.

**Reply by Callahan78** [reply]
dmesg was a good hint . . . there was an undervolting reported. Obviously the power supply was faulty. It has 2.4A and I replaced with another 2.4A power supply and the error message went away. (To be sure I ordered a 3A power supply)
Unfortunately the flickering behaviour doesn´t change.

current dsmeg result:
[dmesg.txt](https://github.com/awawa-dev/HyperHDR/files/7018162/dmesg.txt)


> And don't test with LED strip on the roll please ;) It's a very easy way to make a short circuit.

It´s the silicone isolated version, should be quite safe. But I am working on it . . . I hope to get the frame finished coming weekend so that I can apply the strip. Before I just wanted to be sure, that it will work.
Maybe (as a plan B) I am thinking about a direct-to-Raspberry-solution without ESP and APA102 strip.

**Reply by awawa-dev** [reply]
Did you disable power limit in WLED? 
If the LED strip is working over Wifi then you have got a good connection ESP to LED strip and LED is also not damaged: that's a lot. You should not give up with your current setup.

Maybe HyperHDR logs will shows more or maybe it's an electrical issue (no common ground in fact when connected to Rpi?)
If the WLED is working well over Wifi, the ESP is still connected to the RPi (just to get a power supply)?  Should try that for a test.

**Reply by Callahan78** [reply]
> Did you disable power limit in WLED?

Can´t find this option, just "Enable automatic brightness limiter", here I tried both.

At the moment there are no entries in HyperHDR-Log, I will switch to debug level and let it run a while.




> You should not give up with your current setup.

That would be my favorite :-)



> Maybe HyperHDR logs will shows more or maybe it's an electrical issue (no common ground in fact when connected to Rpi?)
> If the WLED is working well over Wifi, the ESP is still connected to the RPi (just to get a power supply)? Should try that for a tes

Yes, I leave everything connected as it is (ESP gets power from Raspberry in both scenarios), I just switch in HyperHDR between Wifi and USB.

**Reply by awawa-dev** [reply]
Yes, it's a brightness limiter.
You must restart HyperHDR after you changed the log level settings. Frankly I'm out of idea now because adalight in such a simply, basic configuration must work...or at least it should work ;) There is a very slim possibility of conflict with other device over serial port (depending on the system configuration), but it's very rare and difficult situation.

**Reply by awawa-dev** [reply]
Do you have an opportunity to test it from a HyperHDR running on PC or laptop?

**Reply by Callahan78** [reply]
> Yes, it's a brightness limiter.

Ok, I´ll leave it switched off.



> You must restart HyperHDR after you changed the log level settings.

will do



> or at least it should work ;)

yes, sounds very easy :)
For me it seemes that the ESP didn´t get the right data over USB (what is working over Wifi) . . . could it be a configuration thing in HypherHDR?


> There is a very slim possibility of conflict with other device over serial port

Maybe I could try a Raspberry 3 instead 4 . . . (?)



> Do you have an opportunity to test it from a HyperHDR running on PC or laptop?

I have many opportunities :-)
What should I do?


I am out for today, will continue tomorrow . . . thank you VERY much for the great support! 👍

**Reply by awawa-dev** [reply]
It's hard to provide completely wrong data that can be accepted by the other side because adalight has CRC check for a header (data for led aren't protected though). Before jumping to other possible solution logs are necessary ;)

**Comment by Callahan78** [comment]
Here are some logs:
[HyperHDR_Log.txt](https://github.com/awawa-dev/HyperHDR/files/7023349/HyperHDR_Log.txt)

By the way, the colors are not all wrong, some of them seem to be correct.

**Reply by awawa-dev** [reply]
Logs looks OK and now I know device path /dev/ttyUSB0 which will be useful later. Please reduce LED count to 100 LED. We can handle your 300LED later (it's a overkill for adalight at 115200).
If you set a static color from HyperHDR remote control that part of LED strip is OK?

**Comment by Callahan78** [comment]
Setting static color from HyperHDR works (even with 300 LEDs), all LEDs same color, no flickering

Reducing to 100 LEDs:
no flickering anymore. So 300 LEDs seems indeed too much :-p
(at the end I will need around 220 for my 55" TV, I guess)

**Comment by Callahan78** [comment]
device path /dev/ttyUSB0 will change if I choose another USB port?

**Comment by awawa-dev** [comment]
Device path should remain the same if you switch to other USB port.

I run 270 leds using adalight at 2000000 speed (before switching to HyperSPI) so this is not a problem as you have ch340g now. 
Can't exclude completely that one of LED in your LED strip is broken and refuses to work under certain circumstances and all the segment behind is affected....personally had that experience multiple times with one of my SK6812 cold white and I needed to repair it. But that can wait.

Now as you have a working setup for the 100 LEDs try to flash nodemcuv2_ESP8266.bin from https://github.com/awawa-dev/HyperSerialWLED/releases/tag/v3.0.0.0
Set speed 2000000 and enable AWA protocol. We want to see that 100 LEDs working again but at higher speed.

**Reply by Callahan78** [reply]
Ok, flashing is done.
Now I am at the initial problem that HyperHDR does not connect to ESP via USB.

Log:
[HyperHDR_Log_2.txt](https://github.com/awawa-dev/HyperHDR/files/7023582/HyperHDR_Log_2.txt)

**Reply by awawa-dev** [reply]
'Inappropriate ioctl for device' but it was for 115200 which is correct  before you switch to 2000000 ...
And later in the logs? Are there any errors for 2000000?

**Reply by Callahan78** [reply]
There came no more important entries, only these:
"[V4L2:/DEV/VIDEO0] (INFO) Video FPS: 30.03, av. delay: 14ms, good: 1802, bad: 0 (60.02,15)"
nothing else

If the connect works with WLED 0.13 and not with HyperSerialWLED it must be a wrong configuration . . . (?)

**Reply by Callahan78** [reply]
dmesg shows some errors:
https://pastebin.com/rffcy406

**Reply by awawa-dev** [reply]
Configuration looks OK for me. But we can check it. It's a more complicated procedure. We must also switch to HyperSerialESP8266 as it has some output on the serial port.  
1) HyperHDR MUST NOT BE RUNNING  
sudo systemctl stop hyperhdr@pi.service  
test it with a WWW page: HyperHDR page should not be accessible (or test it with ````ps```` command, process must not be running)    
2) Install screen:  
````sudo apt install screen````  
3) flash one of RGBW firmware from https://github.com/awawa-dev/HyperSerialEsp8266/releases/tag/v4.0.0.0  
4) from a command line on RPi (ESP is connected, HyperHDR is not running):  
```` screen /dev/ttyUSB0 2000000````  (to exit screen press: control-a, then key: k, then key: y)
if after one minute there is no output then for sure ESP is not capable for 2000000 speed for some reason...

**Reply by awawa-dev** [reply]
Ups...ch341-uart that something new...new met it before

**Reply by awawa-dev** [reply]
never mind, my 340 has similar logs and is using 341 driver:  
[    7.187383] usbcore: registered new interface driver usbserial_generic
[    7.187468] usbserial: USB Serial support registered for generic
[    7.191974] usbcore: registered new interface driver ch341
[    7.192128] usbserial: USB Serial support registered for ch341-uart
[    7.192272] ch341 1-1.3:1.0: ch341-uart converter detected
[    7.210394] usb 1-1.3: ch341-uart converter now attached to ttyUSB0
but there is one difference about not supporting break control.  
To which port did you connect it? USB2.0 or USB3.0?

**Reply by Callahan78** [reply]
I tried both USB2.0 and USB3.0

But I can try each of the 4 ports

**Reply by awawa-dev** [reply]
No need, seems it's a kind of subfamily of the real ch341...with some hardware limitation that probably break things at higher speed.
OK, you can still proceed with HyperSerialESP8266 but now I have doubts if it is worth...

If 2000000 speed is out of reach we can try a special build of HyperSerialWLED at 921600 (can compile it tomorrow, that 341 is still a mystery but will work for sure with your cp2102), or you can go with a new HyperSPI or to wire the LED strip directly to Rpi...

**Reply by Callahan78** [reply]
I don´t get it why they send me not a CH340 . . . in the description/order it was definitely a CH340 :-/
I would prefer not having a special build because it always makes problems with updates etc.
Maybe I should try a ESP which has CH340 (easiest way).

SK6812 directly to RPi is not possible, as far as I know. (?)

**Reply by awawa-dev** [reply]
With Chinese clones there is always a lottery. If it's 340g then why they removed the markings from the chip and you can't verify it?   
What is a bit funny old 340g was one of the cheapest solution available on the market so they use it almost everywhere.  But maybe due to lack of supplies (Corona) they try new solutions?
  
It's possible to wire SK6812 directly to Rpi but not recommended: for around 220 leds if you won't be running any additional CPU hungry application beside HyperHDR it may succeed.

**Reply by Callahan78** [reply]
it doesn't seem to be that easy to get the right chip at short term . . . maybe we can give it a try with special build on cp2102. If it should working, I will never touch it :-p

**Reply by awawa-dev** [reply]
That's what I like: programmers' golden rule  :)  I'll build it tomorrow.

**Reply by Callahan78** [reply]
> for around 220 leds if you won't be running any additional CPU hungry application beside HyperHDR it may succeed.

The RPi 4 is dedicated for HyperHDR, nothing else.

**Reply by Callahan78** [reply]
> That's what I like: programmers' golden rule :) I'll build it tomorrow.

:-D
Ok, let´s try tomorrow. Thanks! 👍

**Reply by Callahan78** [reply]
In the meantime I could test the wire-directly-version . . . has there only the LEDs data wire to be connected to RPi-GPIO?

**Reply by awawa-dev** [reply]
Yes, if you have a common ground then it's only one wire.

**Reply by Callahan78** [reply]
It´s not, because RPi has its own power supply . . . . so also connect GRD from strip to GPIO?

**Reply by awawa-dev** [reply]
Absolutely, ground from the LED strip to the Rpi's GPIO ground (there are multiple ground pins there).

**Reply by Callahan78** [reply]
Don´t get it working :-p
Connected ground/LED to ground/RPi and tried data/LED to all SPI Pins . . . LEDs stays off . . .
(tested with /dev/spidev0.0 and /dev/spidev0.1, SPI is enabled in raspi-config)

I´ll wait for the special build ;-)

**Reply by awawa-dev** [reply]
Better results for SK6812 are achieved with ws281x rpi pwm using RGBW option (yeah, that's kind of misleading anyway without any guarantee). But HyperHDR must be run as root (https://github.com/awawa-dev/HyperHDR/issues/52).

**Reply by Callahan78** [reply]
Tried, but don´t got the LEDs shining :-p
(LED-Data to GPIO18, LED-ground to GPIO-ground, HyperHDR: ws281x with RGBW-Protocol)

**Reply by awawa-dev** [reply]
Here you go:
![image](https://user-images.githubusercontent.com/69086569/130320381-2cd1b68a-4989-4ad5-9b16-7f3620b97d33.png)
  
Compiled and tested with my Nodemcu v2 cp2102 at speed 921600 with AWA protocol enabled and it works.
[HyperSerialWLED_special_version_speed_921600_nodemcuv2_ESP8266.zip](https://github.com/awawa-dev/HyperHDR/files/7025415/HyperSerialWLED_special_version_speed_921600_nodemcuv2_ESP8266.zip)
  
[HyperSerialWLED_special_version_speed_921600_nodemcuv2_ESP32.zip](https://github.com/awawa-dev/HyperHDR/files/7025457/HyperSerialWLED_special_version_speed_921600_nodemcuv2_ESP32.zip)

**Reply by Callahan78** [reply]
You´re the greatest! :-)
Worked on the first try with my ESP32 👍 

Later I´ll try with the ESP8266. What would you recommend to use in my final setup, the ESP32 or the ESP8266?

Finally I can start with cutting the stripes and soldering the corners :-)

**Reply by awawa-dev** [reply]
ESP32 has a lot more of memory and it's faster which can be a benefit for a greater number of leds in the strip.  
But Wifi connection stability can be more problematic (if you decide to use it anyway).   
  
Because it's 921600 instead of 2000000 I recommend to reduce final refresh rate to around 40 for 220 leds sk6812.  
You can test it yourself later: set refresh rate in the smoothing to 100, set continues output enabled and then in WLED panel check your real refresh rate. Don't go above that value.

**Reply by Callahan78** [reply]
Then I prefer the ESP32 with more performance. If there are any problems with Wifi I could replace by ESP8266. But I don´t plan any purposes via Wifi but configuration. Maybe I´ll try to get an ESP with CH340 later, so I could use the official builds and settings.
Now I have to apply the strip to my aluminium frame. Then I know the exact count of LEDs and I will play around with the refresh rates.
Will get back, when I am ready :-)

**Reply by awawa-dev** [reply]
Looks like a good plan 👍 Good luck then. Sorry to learn that the 340G are counterfeited now. Anyway for your ESP8266 there is always new option https://github.com/awawa-dev/HyperSPI which should give around 80FPS but maybe you should finish your current build first ;-)

**Reply by Callahan78** [reply]
It´s done :-)
Stripes are applied and soldered and the first impression is good. Some colors seemes to be too bright but I think it´s time for fine tuning now.
And I try to understand the HyperSPI :-p

**Reply by awawa-dev** [reply]
SK6812 usually doesn't need a calibration...but there is one important aspect. Colors are represented by red,green,blue and at some point we need to calculate the white channel specially for sk6812. WLED does it its way, HyperSerialEsp8266 has its own way ;) If the LED are just too bright then switch to the 'HyperHDR classic calibration' in the 'processing tab' and decrease a bit luminescence.  

HyperSPI is at very dynamic stage: just as we talk it received support for ESP32 :) Although I use it almost every day for my TV we need do more test for other configuration before we could recommend it for everyone (testers are welcomed of course).

**Reply by Callahan78** [reply]
> If the LED are just too bright then switch to the 'HyperHDR classic calibration' in the 'processing tab' and decrease a bit luminescence.

Does luminescence only affect the white channel? I think that the other colors are overpowered by the white.

At the moment I have another problem. Sometimes, HyperHDR loses the connection to the grabber (I guess, but not sure), then the LEDs show the HyperHDR-Idle-Pattern. After rebooting the RPi, everything ist ok. I'm going to monitor this behaviour a little bit more.

HyperSPI is not available in version 16, correct?
Sounds interesting, but would not fit with my current hardware setup. My Raspberry is in a cupboard and connected to the ESP by a 3m USB cable. I would have to bring the RPi, Grabber and HDMI Splitter directly behind the TV near to the ESP. It´s a little cabling work, but i will think about it :-)

**Reply by awawa-dev** [reply]
> Does luminescence only affect the white channel? I think that the other colors are overpowered by the white.

Ok, there is no universal RGB to RGBW conversion (it's done outside HyperHDR): you can try some other option in WLED (there are at least 3 or 4 don't remember), I've got mine in HyperSerialESP8266 and now in HyperSPI ;) Of course you can lower RGB with luminescence, then white channel also will go down but don't know whether the effect will be satisfactory. Which version of sk6812 (cold/neutral/warm) do you have?

> At the moment I have another problem. Sometimes, HyperHDR loses the connection to the grabber (I guess, but not sure), then the LEDs show the HyperHDR-Idle-Pattern. After rebooting the RPi, everything ist ok. I'm going to monitor this behaviour a little bit more.

Seems like a problem with a power supply again...when that happened you didn't switch the resolution of the HDMI stream somewhere? The answer will be in log or dmesg.

**Reply by Callahan78** [reply]
> Which version of sk6812 (cold/neutral/warm) do you have?

I have neutral.



> Seems like a problem with a power supply again...when that happened you didn't switch the resolution of the HDMI stream somewhere? The answer will be in log or dmesg.

Hm . . . I have a 3A power supply at the Pi, should be enough . . .
Can´t remember if I did do anything what could have causing this . . . but will I will pay attention from now on.
Anyway a good hint, I will keep an eye on it and check dmesg at next occurrence.

**Reply by awawa-dev** [reply]
I guess you can be limited by the USB cable. There is also a 'bug' in some RPi4 revision that force to use their genuine power supply.

**Reply by Callahan78** [reply]
I can´t imagine that the ESP needs that much power . . . but I will try another (shorter) cable.

By the way: today a new ESP arrived and it is a real CH340 :-)
Flashed with nodemcuv2_ESP8266.bin, set to 2000000 and it works :-)

**Reply by Callahan78** [reply]
Just noticed, that Dolby Vision seemes not working. It is not supported, I guess?

**Reply by awawa-dev** [reply]
> I can´t imagine that the ESP needs that much power . . . but I will try another (shorter) cable. 

no ESP, your USB grabber probably...

> Just noticed, that Dolby Vision seemes not working. It is not supported, I guess?  

It's closed format, even grabbers don't supported. Out of chance...

> By the way: today a new ESP arrived and it is a real CH340 :-)
Flashed with nodemcuv2_ESP8266.bin, set to 2000000 and it works :-)

That's great! :smiley:

**Reply by awawa-dev** [reply]
The weakest part of the power supply system is probably USB-C cable which connects Rpi to the power supply. Or the power supply ...RPi 4 B draws around 1.5A under load + add grabber max. 1A + nodemcu max 800mA  ...but it's only guessing :wink:

**Reply by Callahan78** [reply]
> no ESP, your USB grabber probably...

sure . . . I somehow forgot the grabber :-p . . . seems indeed need some power, it gets quite warm.

At the moment it runs without failures. But I didn't have much time to watch TV last days.
I will continue to observe it.

>It's closed format, even grabbers don't supported. Out of chance...

Too bad, I already thought something like that. Maybe one day a device can scale it down . . .

**Reply by awawa-dev** [reply]
sorry I edited your post instead of reply to it :wink:

> Too bad, I already thought something like that. Maybe one day a device can scale it down . . .
  
Outside the USB video grabbers world, some of AVR can process Dolby...for a price a bit more than HDFury Diva they are a much better choice if you plan a home cinema.

**Reply by Callahan78** [reply]
> sorry I edited your post instead of reply to it

No problem :-)



> Outside the USB video grabbers world, some of AVR can process Dolby...for a price a bit more than HDFury Diva they are a much better choice if you plan a home cinema.

Time will work for us . . . maybe in 1-2 years :-)

**Reply by Callahan78** [reply]
At the moment I am still struggling with the colors . . .
Red is more yellow/orange than red. Yellow is almost white. Blue is good.
If I check colors with RGB order wizard, red and green are perfect. Colors from WLED are also perfect. Live TV looks good.
I tried with all settings I could found (and you mentioned), but I cant get red and yellow correct.

Do you have any idea?

**Reply by Callahan78** [reply]
My current settings:
![image](https://user-images.githubusercontent.com/88541621/131228266-624c6702-fe11-49dc-9147-f7b65c7ac853.png)
![image](https://user-images.githubusercontent.com/88541621/131228289-fd4e42a3-9b3a-4952-a9db-ff266b193c48.png)
![image](https://user-images.githubusercontent.com/88541621/131228314-04686c88-9edc-420c-a555-336cd92029ed.png)
![image](https://user-images.githubusercontent.com/88541621/131228335-f604a200-6228-41aa-b2ab-b624cc8505f4.png)
![image](https://user-images.githubusercontent.com/88541621/131228343-0e208f7b-a129-46ba-a09d-0b36f592e727.png)

**Reply by awawa-dev** [reply]
Did you set that red/yellow colors from the HyperHDR's remote tab or you are referencing to colors while watching movie? First check the colors setting them in the remote tab (leave WLED, it leads to nowhere in HyperHDR calibration or color verification..at least for now). If they are OK then verify the live video preview from HyperHDR if it matches the video on the TV.

**Reply by Callahan78** [reply]
It´s while watching TV. To get it clear, I have some pictures with fullscreen color:

![20210828_213029](https://user-images.githubusercontent.com/88541621/131229308-126cef2e-86ec-44eb-9be1-c99d57608c93.jpg)
![20210828_213034](https://user-images.githubusercontent.com/88541621/131229310-78fa26bd-3468-4d20-a74a-933ddb7eb0e3.jpg)
![20210828_213039](https://user-images.githubusercontent.com/88541621/131229311-75160980-d7b6-456a-bcd2-862ae465ea9f.jpg)
![20210828_213043](https://user-images.githubusercontent.com/88541621/131229312-690dcd2e-2b1e-4da5-8e9c-937c311dac86.jpg)

Pictures are slightly falsified due to smartphone camera. But should help to get an impression what I mean.

Seems to be an issue with the grabber? Saw that red is´nt good in live video preview . . . this is a 255,0,0 red (last photo):
![image](https://user-images.githubusercontent.com/88541621/131229547-b082e342-272b-447f-b7a5-f71532927cf6.png)

**Reply by awawa-dev** [reply]
It's definitely wrong. Did you enabled HDR tone mapping? If so, please disable it and check again that red in the preview.

**Reply by Callahan78** [reply]
That´s it . . .
![image](https://user-images.githubusercontent.com/88541621/131229728-75ce5e55-0a61-4b37-addc-830de897792c.png)

Disabling HDR tone mapping does the trick.

**Reply by Callahan78** [reply]
That means that I have to switch that setting depending on the source (HDR or non-HDR)?

**Reply by awawa-dev** [reply]
So it leaves only two options: you don't have an HDR signal or you have Ezcap 320 (the only known to me grabber with some kind of HDR tone mapping) ;)

**Reply by Callahan78** [reply]
What happens without HDR tone mapping? Could no see a difference on the first look.
So I should try to get an Ezcap 320 and replace my WisFox . . .

**Reply by awawa-dev** [reply]
Without HDR tone mapping  it's pure video stream recorder by the grabber, without any software tuning.

You grabber is receiving SDR signal, no HDR. Alone it's not a problem, every grabber should handle SDR. But the question is why that happens and if your TV is receiving HDR signal (that's the main goal, you probably want best video quality after all). The problem could be:
- in the player/video source: it doesn't produce HDR but the SDR
- the HDMI splitter (or the grabber with a loop, dont know exactly your grabber model and video setup) on the way that signals no support for HDR...so the player fallback to SDR
- your TV doesn't support HDR and so the player fallbacks to SDR

**Reply by Callahan78** [reply]
My setup:

Sources:
VU+ 4k Duo Cable Receiver (1080p/4k)
NVIDIA Shield TV Pro (4k, HDR, Dolby Vision)
other non important devices

via HDMI to:

Denon AVRX1300W AV Receiver

via HDMI to:

FeinTech VSP01222 Splitter

via HDMI 1 to LG OLED 55CX8LB TV
via HDMI 2 to Grabber and RPi ....


Playing HDR-Content from NVIDIA Shield TV is no problem. TV displays that it is HDR. Ambilight seemes to be ok, too.
Except Dolby Vision, no problem for TV, but no Ambilight (as discussed).

**Reply by awawa-dev** [reply]
FeinTech VSP01222 can't do HDR to SDR tone mapping. Denon AVR perhaps, but then TV and the grabber would receive the same SDR signal. But if you are sure that TV is receiving HDR signal, you should not worry about your grabber with the SDR: just disable HDR tone mapping in HyperHDR and it will be fine.

**Reply by Callahan78** [reply]
This confuses me a bit, but I just can say that TV shows a message, when I start a HDR movie and everything looks good:
![20210829_001010](https://user-images.githubusercontent.com/88541621/131232262-f955ccaa-10b0-400e-b1e5-3ad18b1b40b6.jpg)
(same with Dolby Vision).
So I can keep my grabber? :-)
